packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,25 +1,46 @@
+import { getActiveTransaction } from '@sentry/tracing';
+
 import { DataFetchingFunction } from './types';
 
 /**
- * Pass-through wrapper for the original function, used as a first step in eventually wrapping the data-fetching
- * functions with code for tracing.
+ * Create a span to track the wrapped function and update transaction name with parameterized route.
  *
  * @template T Types for `getInitialProps`, `getStaticProps`, and `getServerSideProps`
  * @param origFunction The user's exported `getInitialProps`, `getStaticProps`, or `getServerSideProps` function
  * @param context The context object passed by nextjs to the function
  * @returns The result of calling the user's function
  */
-export async function callOriginal<T extends DataFetchingFunction>(
-  origFunction: T['fn'],
-  context: T['context'],
-): Promise<T['result']> {
-  let props;
-
-  // TODO: Can't figure out how to tell TS that the types are correlated - that a `GSPropsFunction` will only get passed
-  // `GSPropsContext` and never, say, `GSSPContext`. That's what wrapping everything in objects and using the generic
-  // and pulling the types from the generic rather than specifying them directly was supposed to do, but... no luck.
-  // eslint-disable-next-line prefer-const, @typescript-eslint/no-explicit-any
-  props = await (origFunction as any)(context);
-
-  return props;
+export async function wrapperCore<T extends DataFetchingFunction>(options: {
+  origFunction: T['fn'];
+  context: T['context'];
+  route: string;
+  op: string;
```

Comment:
Ok as for the thing Abhi mentioned: I took a look at https://develop.sentry.dev/sdk/event-payloads/span/#attributes, and https://develop.sentry.dev/sdk/performance/span-operations/.

From what's already there, it IMO makes the most sense to have something like `op: "render"`, `description: "[dataFetchingFunction]: [route]"`.

We could also expand the `JS framework` operations by something `nextjs.data`, or `nextjs.data.getinitialprops`. I don't know if we would have to involve other teams in this.

Right now I would just go for the first approach so we don't block this PR and fix stuff up later if necessary.

Reply:
yup, and we can always change it if product requires us to.

Reply:
Sounds good!